File: tcg-java/test/de/kimminich/kata/tcg/syntactic/MoveSugar.java
package de.kimminich.kata.tcg.syntactic;

import de.kimminich.kata.tcg.Action;
import de.kimminich.kata.tcg.Card;
import de.kimminich.kata.tcg.Move;

import java.util.Optional;

public class MoveSugar {
    public static Move move(Optional<Card> card, Action action) {
        return new Move(card, action);
    }

    public static Move noMove() {
        return new Move(Optional.empty(), Action.DAMAGE);
    }

}

File: tcg-kotlin/src/test/kotlin/de/kimminich/kata/tcg/syntactic/MoveSugar.kt
package de.kimminich.kata.tcg.syntactic

import de.kimminich.kata.tcg.Action
import de.kimminich.kata.tcg.Card
import de.kimminich.kata.tcg.Move

object MoveSugar {
    fun move(card: Card, action: Action): Move {
        return Move(card, action)
    }

    fun noMove(): Move {
        return Move(Card(-1), Action.DAMAGE)
    }
}
